src/backend/Pg.h

```diff
@@ -255,12 +258,25 @@ class Pg
     friend class PgQuery;
 
     PgConfig const& config_;
+    boost::asio::io_context::strand strand_;
     bool& stop_;
     std::mutex& mutex_;
 
+    int socketNum_ = -1;
```

Comment:
So I actually need to refactor this. According to the docs 
```
Obtains the file descriptor number of the connection socket to the server. A valid descriptor will be greater than or equal to 0; a result of -1 indicates that no server connection is currently open. (This will not change during normal operation, but could change during connection setup or reset.)
```
So it looks like we only need to acquire the socket during connection setup or reset.

Reply:
Yes, same with connect()

Reply:
So I have made the change to 
1) only acquire socket during connection, since it will not change during normal operation.
According to the docs:
"If PQconnectStart succeeds, the next stage is to poll libpq so that it can proceed with the connection sequence. Use PQsocket(conn) to obtain the descriptor of the socket underlying the database connection"
So as long as PQConnectStart succeeds we should be good to obtain the underlying socket.

2) Use RAII to manage the socket, calling `socket_->cancel()` upon destruction.